### PR TITLE
depend on jersey2-api

### PR DESCRIPTION
**Pull request by jetersen**
to avoid class path conflicts. Currently class path conflicts exist between [gitlab-api plugin](https://github.com/jenkinsci/gitlab-api-plugin) and jira plugin

cc @basil @dcendents

https://github.com/jenkinsci/jersey2-api-plugin/pull/19

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub 
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -224,6 +241,13 @@
       <artifactId>mailer</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>jersey2-api</artifactId>
+      <!-- TODO remove the explicit version number once this has made it into the plugin BOM -->
+      <version>2.35-7</version> <!-- TODO https://github.com/jenkinsci/bom/pull/1022 -->
```
Comment (jetersen) [inline pom.xml:248]: Need release of
https://github.com/jenkinsci/bom/pull/1022

**Comment by jetersen** [comment]
@rantoniuk would you be able to take the HPI and do some sanity testing?

I don't have test setup for Jira 😅

**Review by basil — COMMENTED** [review]
I think @dcendents had a way of reproducing the problem? It would be nice to get confirmation that [the incremental build from this PR](https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/jira/3.8-rc1363.0a_029e634220/) resolves the issue.

File: pom.xml [hunk 2]
```diff
@@ -316,7 +340,7 @@
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
         <artifactId>bom-2.332.x</artifactId>
-        <version>1246.va_b_50630c1d19</version>
+        <version>1280.vd669827e38cd</version> <!-- TODO https://github.com/jenkinsci/bom/pull/1022 -->
```
Comment (basil) [inline pom.xml:343]:
```suggestion
        <version>1289.v5c4b_1c43511b_</version>
```

File: pom.xml (hunk 1 above)
Comment (basil) [inline pom.xml:248]:
```suggestion
```

**Comment by jetersen** [comment]
@rantoniuk I think you can remove the waiting for LTS label.

Nothing has changed from the HPI point of view. Though I appreciate if someone would be able to test it. I don't use Jira 😅

**Review by basil — APPROVED** [review]

**Comment by basil** [comment]
Thanks for doing a smoke test Daniel. From my perspective this is ready to land.

**Comment by rantoniuk** [comment]
It's waiting for @olamy since he asked to stop merging any PRs temporarilly because of preparing a security release.

**Review by rantoniuk — APPROVED** [review]

**Pull request merged**